fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/directory/FDBDirectoryManager.java

```diff
@@ -149,4 +159,26 @@ public static FDBDirectoryManager getManager(@Nonnull IndexMaintainerState state
             return newManager;
         }
     }
+
+    private int getMergeDirectoryCount(@Nonnull IndexMaintainerState state) {
+        final AtomicInteger luceneMergeCount = new AtomicInteger();
+        state.store.getRecordMetaData().getAllIndexes().stream().filter(i -> i.getType().equals(LuceneIndexTypes.LUCENE)).forEach(i -> {
```

Comment:
I believe this calculation results in a merge probability of 1/(number of directories) for each directory instead of 1/(number of directories that need merging).

I'm not sure that that's necessarily a problem, but I figured it was worth mentioning so that we're aware that that's what this does.

Reply:
yeah that is true, but as far as I know there might not be a good way to figure out which directories need to merge at this level or at the overrides of ```ConcurrentMergeScheduler.merge()```. So this might be the best solution we can take now.